src/uu/mv/src/mv.rs

```diff
@@ -325,8 +326,9 @@ fn handle_two_paths(source: &Path, target: &Path, opts: &Options) -> UResult<()>
         });
     }
 
+    let is_source_same_target = is_same_file(source, target).unwrap_or(false);
```

Comment:
I wonder if `is_same_file` is already doing the same checks that we do in the Unix version of `are_hardlinks_to_same_file`. Can we simply use `is_same_file` for both platforms?

Reply:
Yes, I think `is_same_file` is doing the same checks, so it could be used in all platforms, but was not sure to change all of that code before confirmation.